app/uploaders/media_item_uploader.rb

```diff
@@ -0,0 +1,91 @@
+# encoding: utf-8
+
+class MediaItemUploader < CarrierWave::Uploader::Base
+  include CarrierWave::MiniMagick
+  include CarrierWave::MimeTypes
+
+  IMAGE_REGEX = /\Aimage\/.*\z/i
+  VIDEO_REGEX = /\Avideo\/.*\z/i
+
+  # the kind of storage to use for this uploader
+  storage :file
+
+  process :set_content_type
+  process :set_size_and_type_on_model
+  process :set_media_kind_on_model
+  process :set_height_and_width_on_model, if: :image?
+
+  version :thumb, if: :image? do
+    process resize_to_fill: [100, 100]
+  end
+  #TODO: Identify appropriate dimensions for these versions
+  version :small
+  version :medium
+  version :large
+
+  # the directory where uploaded files will be stored.
+  def store_dir
+    owner_type = model.media_attachable_type.underscore
+    owner_id = model.media_attachable_id
+    path = File.join('uploads', Rails.env, owner_type, "#{owner_id}", "#{model.id}")
+    path
+  end
+
+  protected
+  def image?(new_file)
+    new_file.content_type =~ IMAGE_REGEX
+  end
+
+  def video?(new_file)
+    new_file.content_type =~ VIDEO_REGEX
+  end
+
+  def set_size_and_type_on_model
+    model.item_content_type = file.content_type if file.content_type
+    model.item_file_size = file.size
+  end
+
+  def set_media_kind_on_model
+    case model.item_content_type
+    when IMAGE_REGEX
+      model.kind = 'image'
+    when VIDEO_REGEX
+      model.kind = 'video'
+    else
+      model.kind = 'file'
+    end
+  end
+
+  def set_height_and_width_on_model
+    if file && model
+      model.item_width, model.item_height = ::MiniMagick::Image.open(file.file)[:dimensions]
+    end
+  end
+
```

Comment:
By default carrierwave only persists the path to the image, when I needed to opt for it over Paperclip I needed to set and persist them manually. I don't know whether they are or will need to be used, but they were present in the schema doc and most are also present by default in Paperclip do I created them here too.